src/main/java/uk/gov/hmcts/dts/fact/services/search/ServiceAreaSearchFactory.java

```diff
@@ -31,5 +31,23 @@ public Search getSearchFor(final ServiceArea serviceArea, final MapitData mapitD
             default:
                 return defaultSearch;
         }
+
+    }
+
+    public Search getSearchForNearest(final ServiceArea serviceArea, final MapitData mapitData, final String action) {
+
+        if ("nearest".equals(action)) {
```

Comment:
It feels to me that the new getSearchForNearest method is pretty much identical to the one that precedes it, but with an extra if statement 

It also seems like moving fowards all of the business logic should be driven by actions

would it make more sense for us to put the if statement into getSearchFor and for the methods that call it, also include the action passed through the frontend? 

Would remove the need to have another method (food for thought, but let me know if I am barking up the wrong tree here!)